mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxExtensions.kt

```diff
@@ -29,6 +29,40 @@ inline fun <T, reified VM : BaseMvRxViewModel<S>, reified S : MvRxState> T.fragm
         .apply { subscribe(this@fragmentViewModel, subscriber = { postInvalidate() }) }
 }
 
+/**
+ * Gets or creates a ViewModel scoped to a parent fragment. This delegate will walk up the parentFragment hierarchy
+ * until it finds a Fragment that can provide the correct ViewModel. If no parent fragments can provide the ViewModel,
+ * a new one will be created in the direct parent of the curent Fragment.
```

Comment:
It doesn't feel quite right that an existing viewmodel can exist anywhere in the parent fragment hierarchy, but creating a new one also puts it in the direct parent.

What if instead by default it creates it in the top level parent fragment? That seems less likely to cause issues

Reply:
@elihart @BenSchwab and I chatted about this yesterday with the same concern. Do you think the top would be a more sane default than the first parent?

Reply:
Yeah, feels safer since any other child fragment is guaranteed to be able to reach the same viewmodel.

This could be important in an edge case like process recreation where we are starting with a nested child fragment that wouldn't normally be created first

Reply:
@elihart would that be the case for restoration? The ViewModel won't be created until ON_CREATE. Would the child onCreate be called before the parent? That seems impossible because onAttach is called before onCreate.